src/topology/bounded_continuous_function.lean

```diff
@@ -429,5 +429,109 @@ def of_normed_group_discrete {α : Type u} {β : Type v}
   (f : α  → β) (C : ℝ) (H : ∀x, norm (f x) ≤ C) : α →ᵇ β :=
 of_normed_group f C H continuous_of_discrete_topology
 
+/- Formula for norm. -/
+lemma norm_eq ( f : α →ᵇ β) :
+  ∥f∥ = Inf {C : ℝ | 0 ≤ C ∧ ∀ (x : α), ∥f x∥ ≤ C} :=
+begin
+  rw ← sub_zero f,
+  rw ← dist_eq_norm,
+  rw bounded_continuous_function.dist_eq,
+  simp,
+end
+
+
 end normed_group
+
+
+section normed_space
+/- In this section, if β is a normed space, then we show that the space of bounded
+continuous functions from α to β inherits a normed space structure, by using
+pointwise operations and checking that they are compatible with the uniform distance. -/
+
+variables [nondiscrete_normed_field 𝕜]
+variables [topological_space α] [normed_group β] [normed_space 𝕜 β]
+variables {f g : α →ᵇ β} {x : α} {C : ℝ}
+
+
+instance : has_scalar 𝕜 (α →ᵇ β) :=
+  ⟨λc, λf, ⟨λx, c • f.1 x,
+    continuous.smul continuous_const f.2.left,
+    begin
+      cases f.2.right with C hbound,
+      use ∥c∥ * C,
+      intros,
+      have hnneg : 0 ≤ ∥c∥,
+      { exact norm_nonneg c },
+      specialize hbound x y,
+      rw dist_eq_norm at hbound,
+      rw dist_eq_norm,
+      calc ∥c • f x - c • f y∥ = ∥c • (f x - f y)∥ : by rw smul_sub c (f x) (f y)
+      ... = ∥c∥ * ∥f x - f y∥ : norm_smul c (f x - f y)
+      ... ≤ ∥c∥ * C : mul_le_mul_of_nonneg_left hbound hnneg,
+    end⟩⟩
+
+
+instance : module 𝕜 (α →ᵇ β) :=
+  module.of_core $ begin
+    refine { smul := (•), ..},
+    intros c f g, ext,
+    exact smul_add c (f x) (g x),
+    intros c₁ c₂ f, ext,
+    exact add_smul c₁ c₂ (f x),
+    intros c₁ c₂ f,
+    ext, exact mul_smul c₁ c₂ (f x),
+    intros f,
+    ext, exact one_smul 𝕜 (f x),
+  end
```

Comment:
```
module.of_core $
{ smul := (•),
   other_field := by { intros, ext, tactic },
   etc... ..},
```
In VScode, you can write `{! !}` when you are about to write a term for a structure or class. Then a little :bulb: will show up, and you can click it, and select "generate skeleton for structure" or something like that. It will fill in all the field names for you.

Reply:
Fixed, thank you.